lib/rules/prefer-regex-literals.js

```diff
@@ -306,10 +388,69 @@ module.exports = {
 
                 for (const { node } of tracker.iterateGlobalReferences(traceMap)) {
                     if (disallowRedundantWrapping && isUnnecessarilyWrappedRegexLiteral(node)) {
+                        const regexNode = node.arguments[0];
+
                         if (node.arguments.length === 2) {
-                            context.report({ node, messageId: "unexpectedRedundantRegExpWithFlags" });
+                            const suggests = [];
+
+                            const argFlags = getStringValue(node.arguments[1]) || "";
+
+                            if (canFixTo(node, regexNode.regex.pattern, argFlags)) {
+                                suggests.push({
+                                    messageId: argFlags ? "replaceWithLiteralAndFlags" : "replaceWithLiteral",
```

Comment:
In cases like this:

```js
new RegExp(/foo/u, "");
```

I think the message for the equivalent replacement should be `"replaceWithLiteralAndFlags"` (`Replace with an equivalent regular expression literal with flags ''.`) to emphasize that the flags will be empty. I think we can just always use `"replaceWithLiteralAndFlags"` here.

Reply:
@mdjermanovic Thanks :) I fixed it. https://github.com/eslint/eslint/pull/16658/commits/de98a290c4b6842460d2d0ce36d6a93b1865b87b